code-snippet.rkt

```diff
@@ -39,27 +40,37 @@
   
   #:methods gen:custom-write
   [(define (write-proc this out mode)
+     (define start-line (code-snippet-start-line this))
      (define end-line (code-snippet-end-line this))
+     (define end-line-digit-count (digit-count end-line))
+     (define leading-indentation (code-snippet-start-column this))
      (for ([line (in-lines (open-input-string (code-snippet-raw-text this)))]
            [n (in-naturals)])
        (define line-number-string
-         (~a (+ n (code-snippet-start-line this)) #:min-width (digit-count end-line)))
+         (~a (+ n start-line) #:min-width end-line-digit-count))
        (write-string line-number-string out)
        (write-string " " out)
        (cond
          [(zero? n) (write-string line out)]
          [else
-          (define leading-indentation (make-string (code-snippet-start-column this) #\space))
-          (write-string (string-replace line leading-indentation "" #:all? #false) out)])
+          (write-string (substring line leading-indentation) out)])
```

Comment:
Hmm, I'm not totally sure and I haven't tested this, but I think this isn't right if there's a code snippet that contains text in later lines that starts earlier than the first line does. For example, in this:

```
   foo {
bar;
}
```
The leading indentation is 3, but later lines aren't always indented 3 columns, so just taking a substring would drop stuff. Even properly indented Racket code can look like this sometimes because of multiline string literals.